transport/src/main/java/io/el/channel/AbstractChannelHandlerContext.java

```diff
@@ -111,16 +111,74 @@ private AbstractChannelHandlerContext findContextInbound() {
     return ctx;
   }
 
+  private AbstractChannelHandlerContext findContextOutbound() {
+    AbstractChannelHandlerContext ctx = this;
+    do {
+      ctx = ctx.prev;
+    } while (ctx.executionFlag != FLAG_OUTBOUND);
+    return ctx;
+  }
+
+  public ChannelPromise newPromise() {
+    return new DefaultChannelPromise(channel(), eventLoop());
+  }
+
   @Override
   public ChannelPromise bind(SocketAddress localAddress) {
-    // TODO: implement me
-    return null;
+    return this.bind(localAddress, newPromise());
   }
 
   @Override
   public ChannelPromise bind(SocketAddress localAddress, ChannelPromise promise) {
-    // TODO: implement me
-    return null;
+    ObjectUtil.checkNotNull(localAddress, "localAddress");
+    if (isNotValidPromise(promise)) {
+      // canceled
+      return promise;
+    }
+
+    final AbstractChannelHandlerContext next = findContextOutbound();
+    final EventLoop eventLoop = next.eventLoop();
+    if (eventLoop.inEventLoop()) {
+      next.bind(localAddress, promise);
+    } else {
+      safeExecute(eventLoop, () -> {
+        next.bind(localAddress, promise);
+      }, promise);
+    }
+    return promise;
+  }
+
+  private static boolean safeExecute(EventLoop eventLoop, Runnable runnable, ChannelPromise promise) {
+    try {
+      eventLoop.execute(runnable);
+      return true;
+    } catch (Throwable cause) {
+      promise.setFailure(cause);
+      return false;
+    }
+  }
+
+  private boolean isNotValidPromise(ChannelPromise promise) {
+    ObjectUtil.checkNotNull(promise, "promise");
+
+    if (promise.isDone()) {
+      // if the promise is canceled, we do not need to continue the code
+      if (promise.isCancelled()) {
+        return true;
+      }
+      throw new IllegalArgumentException("promise already done: " + promise);
+    }
+
+    if (promise.channel() != channel()) {
```

Comment:
It would be great to add comments on why the channel of promise should be equal with the context's

Reply:
@zeroFruit I added it: https://github.com/zeroFruit/el/pull/54/commits/d562a51389ab1cef954c39128169c97cdcabc4ff